native/packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js

```diff
@@ -35,48 +40,157 @@ const styles = StyleSheet.create({
     borderColor: 'transparent',
     borderBottomColor: tokens.colorGray100,
     borderWidth: tokens.borderSizeSm,
+  },
+  horizontalNavInner: {
     flexDirection: 'row',
   },
   spaceAround: {
     flex: 0,
     flexGrow: 1,
     justifyContent: 'space-around',
   },
+  selectedIndicator: {
+    backgroundColor: tokens.colorBlue700,
+    height: tokens.borderSizeLg,
+  },
 });
 
-const BpkHorizontalNav = (props) => {
-  const {
-    children,
-    spaceAround,
-    style,
-    ...rest
-  } = props;
-  const navStyle = [styles.horizontalNav];
-  if (spaceAround) {
-    navStyle.push(styles.spaceAround);
+const validateSelectedIndex = (selectedIndex, childrenCount) => {
+  if (typeof selectedIndex !== 'number' || !Number.isInteger(selectedIndex)) {
+    return false;
   }
-  return (
-    <ScrollView
-      alwaysBounceHorizontal={false}
-      contentContainerStyle={[navStyle, style]}
-      horizontal
-      showsHorizontalScrollIndicator={false}
-      {...rest}
-    >
-      { children }
-    </ScrollView>
-  );
+  if (selectedIndex < 0 || selectedIndex + 1 > childrenCount) {
+    return false;
+  }
+  return true;
 };
 
-BpkHorizontalNav.propTypes = {
-  children: PropTypes.node.isRequired,
+class BpkHorizontalNav extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      indicatorOffsetX: null,
+      indicatorWidth: null,
+    };
+    this.onChildLayout = this.onChildLayout.bind(this);
+    this.animateIndicator = this.animateIndicator.bind(this);
+  }
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.selected !== null) {
+      this.animateIndicator(this.childrenPositions[nextProps.selected]);
+    }
+  }
+  onChildLayout = (event, index) => {
+    const { width, x } = event.nativeEvent.layout;
+    this.childrenPositions[index] = { width, x };
+
+    // If the child in question is the initially selected one,
+    // the indicator can now be positioned.
+    if (this.props.selected === index) {
+      this.setState({
+        indicatorOffsetX: new Animated.Value(x),
+        indicatorWidth: new Animated.Value(width),
+      });
+    }
+  }
+  animateIndicator = ({ x, width }) => {
+    const duration = 200;
+    Animated.parallel([
+      Animated.timing(this.state.indicatorOffsetX, {
+        toValue: x,
+        duration,
+      }),
+      Animated.timing(this.state.indicatorWidth, {
+        toValue: width,
+        duration,
+      }),
+    ]).start();
+  }
+  childrenPositions = {};
```

Comment:
Could we introduce more vertical spacing in this file. One line between each member of the component for example?